epochStart/metachain/stakingDataProvider_test.go

```diff
@@ -465,16 +466,63 @@ func TestStakingDataProvider_PrepareStakingDataForRewards(t *testing.T) {
 func TestStakingDataProvider_FillValidatorInfo(t *testing.T) {
 	t.Parallel()
 
-	owner := []byte("owner")
-	topUpVal := big.NewInt(828743)
-	basePrice := big.NewInt(100000)
-	stakeVal := big.NewInt(0).Add(topUpVal, basePrice)
-	numRunContractCalls := 0
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
 
-	sdp := createStakingDataProviderWithMockArgs(t, owner, topUpVal, stakeVal, &numRunContractCalls)
+		owner := []byte("owner")
+		topUpVal := big.NewInt(828743)
+		basePrice := big.NewInt(100000)
+		stakeVal := big.NewInt(0).Add(topUpVal, basePrice)
+		numRunContractCalls := 0
 
-	err := sdp.FillValidatorInfo(&state.ValidatorInfo{PublicKey: []byte("bls key")})
-	require.NoError(t, err)
+		sdp := createStakingDataProviderWithMockArgs(t, owner, topUpVal, stakeVal, &numRunContractCalls)
+
+		err := sdp.FillValidatorInfo(&state.ValidatorInfo{PublicKey: []byte("bls key")})
+		require.NoError(t, err)
+	})
+	t.Run("concurrent calls should work", func(t *testing.T) {
+		t.Parallel()
+
+		defer func() {
```

Comment:
[comment]
I think we could've used the `require.NotPanics` func here which takes care of that

Reply:
pushed, I was used with the recover version, good to know